Respect empty scheduler/placement annotations & handle actorStateStore without placement

cicoyle opened the pull request:
This PR fixes two issues related to disabling connections to the scheduler & placement service via annotations.

1. Respect Explicitly Empty Annotations
Users can disable connecting to scheduler and placement per [our docs](https://docs.dapr.io/reference/arguments-annotations-overview/) by setting the annotations field to `""` or `" "`.
Currently, the sidecar injector overwrites these values with the k8s address instead of respecting them.
Now, the sidecar injector properly respects `""` or `" "` for both placement & scheduler addresses.

2. Handle `actorStateStore=true` with Placement Disabled
If `actorStateStore: true` is set but placement is explicitly disabled, the previous behavior resulted in errors like:
```
time="2025-02-07T14:55:50.519322722Z" level=debug msg="api error: code = Internal desc = dapr is not ready"
time="2025-02-07T14:55:50.892991847Z" level=error msg="Failed to connect to placement: rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp: address \\\"\\\": missing port in address\""
```

```
time="2025-02-07T15:35:07.952844299Z" level=info msg="Actor runtime disabled: api error: code = Internal desc = the state store is not configured to use the actor runtime."
```
Now, if `actorStateStore=true` but placement is disabled, Dapr will simply log that placement is disabled.

I also explicitly log about if placement is disabled that actors + workflow api cannot be used along with the scheduler not being connected when configured.

[Should fix this issue](https://github.com/dapr/dapr/issues/8100)

JoshVanL commented on pkg/runtime/config.go:
```diff
@@ -317,6 +318,17 @@ func (c *Config) toInternal() (*internalConfig, error) {
+			break
+		}
+	}
+
```
Why wouldn’t a non valid placement address not error?

cicoyle replied:
The double negative is confusing me, but I did double check tho that this is no longer needed with the injector respecting the user annotations now for `""` && `" "`. removed the check

JoshVanL commented on pkg/runtime/scheduler/scheduler.go:
```diff
@@ -82,8 +83,10 @@ func New(opts Options) *Scheduler {
 func (s *Scheduler) Run(ctx context.Context) error {
-	if len(s.addresses) == 0 {
+	if len(s.addresses) == 0 ||
+		len(s.addresses) == 1 && strings.TrimSpace(s.addresses[0]) == "" {
```
```suggestion
		(len(s.addresses) == 1 && strings.TrimSpace(s.addresses[0]) == "") {
```

JoshVanL commented on pkg/runtime/wfengine/wfengine.go:
```diff
@@ -142,6 +143,13 @@ func (wfe *engine) RegisterGrpcServer(server *grpc.Server) {
 
 func (wfe *engine) Run(ctx context.Context) error {
+	if wfe.actors.RuntimeStatus().GetRuntimeStatus() == runtimev1pb.ActorRuntime_DISABLED {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		}
```
```suggestion
  <-ctx.Done()
  return ctx.Error()
```

JoshVanL commented on tests/integration/suite/daprd/actors/actors.go:
```diff
@@ -0,0 +1,71 @@
+)
+
+func init() {
+	suite.Register(new(actors))
```
This should live under `/suite/actors`. That package itself is probably in the wrong place and should live under `/suite/daprd/actors` but we shouldn't change that in this PR.

Rename this test (struct) to `emptyaddress` or similar.

cicoyle replied:
yeah I agree that the `/suite/actors` should move to `/suite/daprd/actors`. For now, I've changed the struct to be `emptyaddress` per your suggestion.

JoshVanL commented on tests/integration/suite/daprd/actors/actors.go:
```diff
@@ -0,0 +1,71 @@
+/*
```
👍

JoshVanL reviewed (changes requested):
@cicoyle after we've move the int test file under `/suite/actors` we can merge.

cicoyle commented:
> @cicoyle after we've move the int test file under `/suite/actors` we can merge.

I thought it was fine for the new test I added to be in the proper path and a follow up PR would move the `suite/actors` tests to the proper dir that I added a test to which is `suite/daprd/actors`, but that would be in a separate PR and not in my PR here

cicoyle commented:
cc @JoshVanL I opened[ this PR](https://github.com/dapr/dapr/pull/8481) to move those tests to the proper path. I would appreciate if this current PR did not have to wait since I opened the other PR to move the tests :)

JoshVanL reviewed (approved):
Let’s make sure to rename the added int test file to `emptyaddress`.
